src/wp-includes/view-transitions.js

```diff
@@ -0,0 +1,191 @@
+				}
+				if ( oldPageMatches && Number( oldPageMatches[ 1 ] ) > 1 ) {
+					return `${ prefix }backwards`;
+				}
+			}
+		}
+
+		// Check if the URLs are for content labelled by date (e.g. navigation to previous/next post).
+		const oldDateMatches = oldPathname.match( /\/(\d{4})\/(\d{2})\/(\d{2})\/[^\/]+\/?$/ );
+		const newDateMatches = newPathname.match( /\/(\d{4})\/(\d{2})\/(\d{2})\/[^\/]+\/?$/ );
+		if ( oldDateMatches && newDateMatches ) {
+			const oldPageBase = oldPathname.substring( 0, oldPathname.length - oldDateMatches[ 0 ].length );
+			const newPageBase = newPathname.substring( 0, newPathname.length - newDateMatches[ 0 ].length );
+			if ( oldPageBase === newPageBase ) { // They belong to the same hierarchy.
+				const oldDate = new Date( parseInt( oldDateMatches[ 1 ] ), parseInt( oldDateMatches[ 2 ] ) - 1, parseInt( oldDateMatches[ 3 ] ) );
+				const newDate = new Date( parseInt( newDateMatches[ 1 ] ), parseInt( newDateMatches[ 2 ] ) - 1, parseInt( newDateMatches[ 3 ] ) );
+				if ( oldDate < newDate ) {
+					return 'forwards';
+				}
+				if ( oldDate > newDate ) {
+					return 'backwards';
+				}
+			}
+		}
+
+		return 'default';
+	};
+
+	window.addEventListener( 'pageswap', ( e ) => {
+		if ( e.viewTransition ) {
+			const transitionType = determineTransitionType( e.activation.from, e.activation.entry );
+			e.viewTransition.types.add( transitionType );
+
+			if ( document.body.classList.contains( 'single' ) ) {
+				setTemporaryViewTransitionNames(
+					getViewTransitionEntries(
+						transitionType,
+						document.body,
+						getArticle()
+					),
+					e.viewTransition.finished
+				);
+			} else if ( document.body.classList.contains( 'home' ) || document.body.classList.contains( 'archive' ) ) {
+				setTemporaryViewTransitionNames(
+					getViewTransitionEntries(
+						transitionType,
+						document.body,
+						getArticleForUrl( e.activation.entry.url )
+					),
+					e.viewTransition.finished
+				);
+			}
+		}
+	} );
+
+	window.addEventListener( 'pagereveal', ( e ) => {
+		if ( e.viewTransition ) {
+			const transitionType = determineTransitionType( window.navigation.activation.from, window.navigation.activation.entry );
+			e.viewTransition.types.add( transitionType );
+
+			if ( document.body.classList.contains( 'single' ) ) {
+				setTemporaryViewTransitionNames(
+					getViewTransitionEntries(
+						transitionType,
+						document.body,
+						getArticle()
+					),
+					e.viewTransition.ready
+				);
+			} else if ( document.body.classList.contains( 'home' ) || document.body.classList.contains( 'archive' ) ) {
+				setTemporaryViewTransitionNames(
+					getViewTransitionEntries(
+						transitionType,
+						document.body,
+						window.navigation.activation.from ? getArticleForUrl( window.navigation.activation.from.url ) : null
+					),
+					e.viewTransition.ready
+				);
+			}
+		}
+	} );
+} else {
+	window.console.warn( 'View transitions not loaded as the browser is lacking support.' );
```

Comment:
I assume this is for debugging?

Reply:
It's there to emit a warning if the browser doesn't support the feature. Obviously not something that's important to surface to the end user, but I think it's worthwhile to have even in production code so that it's easy for a developer to spot whether the feature _should_ be working in the current environment or not.